package/com.unity.formats.usd/Runtime/Scripts/IO/SanitizedSamples.cs

```diff
@@ -0,0 +1,473 @@
+
+            if (primvar.value.GetType() == typeof(Vector3[]))
+            {
+                var value = primvar.value as Vector3[];
+                ConvertInterpolationToFaceVarying(ref value, faceVertexIndices, changeHandedness, primvar.GetInterpolationToken());
+                primvar.SetValue(value);
+                return;
+            }
+
+            if (primvar.value.GetType() == typeof(Vector4[]))
+            {
+                var value = primvar.value as Vector4[];
+                ConvertInterpolationToFaceVarying(ref value, faceVertexIndices, changeHandedness, primvar.GetInterpolationToken());
+                primvar.SetValue(value);
+                return;
+            }
+        }
+
+        static void FlattenUVs(Primvar<object> primvar)
+        {
+            if (primvar.value == null)
+                return;
+
+            // We can't use uv.GetValueType() as it return "typeof(T)" and so would return "object" in this case instead of
+            // the actual type of value.
+            if (primvar.value.GetType() == typeof(Vector2[]))
+            {
+                var value = primvar.value as Vector2[];
+                Flatten(ref value, primvar.indices);
+                primvar.SetValue(value);
+                return;
+            }
+
+            if (primvar.value.GetType() == typeof(Vector3[]))
+            {
+                var value = primvar.value as Vector3[];
+                Flatten(ref value, primvar.indices);
+                primvar.SetValue(value);
+                return;
+            }
+
+            if (primvar.value.GetType() == typeof(Vector4[]))
+            {
+                var value = primvar.value as Vector4[];
+                Flatten(ref value, primvar.indices);
+                primvar.SetValue(value);
+                return;
+            }
+        }
+
+        internal static void TriangulateAttributes<T>(ref T[] values, int[] faceVertexCount, bool changeHandedness)
+        {
+            var newValues = new List<T>();
+            var last = 0;
+            for (var i = 0; i < faceVertexCount.Length; i++)
+            {
+                var next = last + 1;
+                for (var t = 0; t < faceVertexCount[i] - 2; t++)
+                    if (changeHandedness)
+                    {
+                        newValues.Add(values[next++]);
+                        newValues.Add(values[last]);
+                        newValues.Add(values[next]);
+                    }
+                    else
+                    {
+                        newValues.Add(values[last]);
+                        newValues.Add(values[next++]);
+                        newValues.Add(values[next]);
+                    }
+
+                last += faceVertexCount[i];
+            }
+
+            values = newValues.ToArray();
+        }
+
+        void ConvertInterpolationToFaceVarying<T>(ref T[] values, int[] vertexIndices, bool changeHandedness = false, TfToken interpolation = null)
```

Comment:
I think it's worth adding a doc string for this function, since converting TO FaceVarying for import sounds strange.

I would expect things to be converted to Vertex/Varying interpolation for import.

Reply:
Added in dd63d81